### PR TITLE
Feature/Expand IdentityDVO from incoming Request if available

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**

### CODE REVIEW

**Comment by jkoenig134** [comment]
The idea was to not expand from a RelationshipTemplate and instead to expand from the open incoming Request.

File: packages/runtime/test/dataViews/IdentityDVO.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    templator.eventBus.reset();
+});
+
+describe("IdentityDVO after loading a relationship template sharing a DisplayName", () => {
+    beforeAll(async () => {
+        const senderAttribute = await templator.consumption.attributes.createRepositoryAttribute({
+            content: {
+                value: {
+                    "@type": "DisplayName",
+                    value: "Dr. Theodor Munchkin von Reichenhardt"
```
Comment (jkoenig134) [inline packages/runtime/test/dataViews/IdentityDVO.test.ts:29]: "A Display Name"

File: packages/runtime/test/dataViews/IdentityDVO.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        templatorTemplate = await createTemplate(templator.transport, templateContent);
+    });
+    test("IdentityDVO should use the DisplayName", async () => {
+        const requestorTemplate = (await requestor.transport.relationshipTemplates.loadPeerRelationshipTemplate({ reference: templatorTemplate.truncatedReference })).value;
+
+        const dto = requestorTemplate;
+        const dvo = (await requestor.expander.expandRelationshipTemplateDTO(dto)) as PeerRelationshipTemplateDVO;
+        expect(dvo).toBeDefined();
+        expect(dvo.createdBy.name).toBe("Dr. Theodor Munchkin von Reichenhardt");
+        expect(dvo.createdBy.initials).toBe("DTMvR");
```
Comment (jkoenig134) [inline packages/runtime/test/dataViews/IdentityDVO.test.ts:57]: also check peer

Comment (Magnus-Kuhn) [reply packages/runtime/test/dataViews/IdentityDVO.test.ts:57]: what do you mean? Letting the peer expand his own address? The templator expanding the requestor's address could be a thing once personalized templates are there

Comment (jkoenig134) [reply packages/runtime/test/dataViews/IdentityDVO.test.ts:57]: wait what? `dvo.peer` ist the same identity as `dvo.createdBy` the peer should be expanded in the exactly same way

All I want you to do is putting 

```
expect(dvo.peer.name).toBe("Dr. Theodor Munchkin von Reichenhardt");
expect(dvo.peer.initials).toBe("DTMvR");
```

as checks to this test.

If this doesn't work your expander doesn't work.

Comment (Magnus-Kuhn) [reply packages/runtime/test/dataViews/IdentityDVO.test.ts:57]: dvo.peer doesn't exist on PeerRelationshipTemplateDVO

Comment (jkoenig134) [reply packages/runtime/test/dataViews/IdentityDVO.test.ts:57]: Oooh, you are expanding a relationshipTemplate here? I didn't got that context. Please add another test where you expand a LocalRequest to a LocalRequestDVO where you can check `dvo.peer` AND `dvo.createdBy`. That's the usecase in the app.

File: packages/runtime/src/dataViews/DataViewExpander.ts [hunk 3]
```diff
@@ -1588,11 +1588,24 @@ export class DataViewExpander {
...
+                query: {
+                    peer: address,
+                    status: [LocalRequestStatus.ManualDecisionRequired, LocalRequestStatus.DecisionRequired]
+                }
+            })
+        ).value;
+        requestResult.sort((request1, request2) => (request2.createdAt > request1.createdAt ? 1 : -1));
+        if (requestResult.length > 0) {
+            return this.expandAddressFromRequest(requestResult.at(-1)!);
+        }
```
Comment (jkoenig134) [inline packages/runtime/src/dataViews/DataViewExpander.ts:1606]: we know there is no relationship - there will only be 0 or 1 request available.

Comment (Magnus-Kuhn) [reply packages/runtime/src/dataViews/DataViewExpander.ts:1606]: it is for the case that multiple different templates have been loaded previously, each generating their own request

Comment (jkoenig134) [reply packages/runtime/src/dataViews/DataViewExpander.ts:1606]: nope, we block these other loads.

Comment (Magnus-Kuhn) [reply packages/runtime/src/dataViews/DataViewExpander.ts:1606]:
```
await exchangeTemplate(sRuntimeServices.transport, rRuntimeServices.transport, templateContent);
await exchangeTemplate(sRuntimeServices.transport, rRuntimeServices.transport, templateContent);
const requestsResult = await rRuntimeServices.consumption.incomingRequests.getRequests({});
```
(tried it in the RequestModule.test around line 100) results in a length 2 array

Comment (jkoenig134) [reply packages/runtime/src/dataViews/DataViewExpander.ts:1606]: this is **really** bad. I'll look into it.

Comment (Magnus-Kuhn) [reply packages/runtime/src/dataViews/DataViewExpander.ts:1606]: To not have this conversation open forever: I assume now that there is max 1 request.

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**